### PR TITLE
Optimize custom input

### PR DESCRIPTION
**Pull request by pingan1927**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [ ] Rebase before creating a PR to keep commit history clear.
* [ ] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1736)

### CODE REVIEW

**Comment by silentcloud** [comment]
统一用这个 mixin？ https://github.com/react-component/util/blob/master/src/getContainerRenderMixin.jsx

File: components/input-item/CustomInput.web.tsx [hunk 1]
```diff
@@ -50,6 +54,35 @@ class NumberInput extends React.Component<any, any> {
...
+    />);
+  }
+
+  renderCustomKeyboard = () => {
+    let container = document.querySelector(`#${this.props.keyboardPrefixCls}-container`);
+    if (!container) {
+      container = document.createElement('div');
+      container.setAttribute('id', `${this.props.keyboardPrefixCls}-container`);
+      document.body.appendChild(container);
+      (window as any).antmCustomKeyboard = ReactDOM.unstable_renderSubtreeIntoContainer(
```
Comment (silentcloud) [inline components/input-item/CustomInput.web.tsx]: 统一用这个 mixin？ https://github.com/react-component/util/blob/master/src/getContainerRenderMixin.jsx

https://github.com/react-component/dialog/blob/master/src/DialogWrap.tsx

Comment (pingan1927) [reply components/input-item/CustomInput.web.tsx]: 本来是想直接引入这个的，想想还是算了，区别挺大，仅仅都是使用unstable_renderSubtreeIntoContainer

**Comment by paranoidjk** [comment]
另外可以随 PR 一起补上 changelog。


等 PR 被合并之后，记得手动同步到 master。

File: components/input-item/CustomInput.web.tsx [hunk 2]
```diff
@@ -66,35 +100,66 @@ class NumberInput extends React.Component<any, any> {
...
       } else {
-        onChange({ target: { value: (value + KeyboardItemValue) } });
+        valueAfterChange = (value + KeyboardItemValue);
+        onChange({ target: { value: valueAfterChange } });
       }
     }
+
+    const antmCustomKeyboard = (window as any).antmCustomKeyboard;
+    antmCustomKeyboard.confirmDisabled = (valueAfterChange === '');
+    if (valueAfterChange === '') {
```
Comment (paranoidjk) [inline components/input-item/CustomInput.web.tsx]: 这个逻辑为什么不在 custom keyboard 内部处理掉？

File: components/input-item/CustomInput.web.tsx [hunk 3]
```diff
@@ -66,35 +100,66 @@ class NumberInput extends React.Component<any, any> {
...
+      }
     });
-    this.props.onFocus(value);
   }
 
   onKeyboardClick = (KeyboardItemValue) => {
     let { value, onChange, maxLength } = this.props;
+    let valueAfterChange;
     // 删除键
     if (KeyboardItemValue === 'delete') {
```
Comment (paranoidjk) [inline components/input-item/CustomInput.web.tsx]: 太多 if else 嵌套了，这个逻辑更适合 Guard Clause

File: components/input-item/CustomInput.web.tsx [hunk 4]
```diff
@@ -50,6 +55,35 @@ class NumberInput extends React.Component<any, any> {
       clearTimeout(this.debounceFocusTimeout);
       this.debounceFocusTimeout = null;
     }
+    const antmCustomKeyboard = (window as any).antmCustomKeyboard;
+    if (antmCustomKeyboard.linkedInput === this) {
```
Comment (paranoidjk) [inline components/input-item/CustomInput.web.tsx]: 这个抽象一个方法出来？不少地方都是判断 linkedInput 是否是 this 再执行一些逻辑

**Comment by paranoidjk** [comment]
另外，所有使用到 window 的场景，留心下不要给服务端渲染造成麻烦

**Comment by pingan1927** [comment]
都在didMount之后才会处理window.

File: components/input-item/CustomInput.web.tsx (hunk 2 above)
Comment (pingan1927) [reply components/input-item/CustomInput.web.tsx]: 样式方面的变化，custom keyboard本身实例化之后不好处理。
思路已经变成通过钩子(type="money")实例化出唯一一个全局自定义键盘，再通过钩子(onFocus)关联到某一个随机的带自定义键盘的输入框。特定input的onChange是没法通过react的render机制重新渲染自定义键盘。
自定义键盘的功能一旦实例化后，后续的样式、事件触发都通过修改实例的属性以及addClass/reomveClass来实现。

**Pull request merged**